### ISSUE TITLE
weed shell - can't delete volume from server

### ISSUE THREAD

**Issue by ignusws** [chrislusf/seaweedfs#1283]
Sponsors SeaweedFS via Patreon https://www.patreon.com/seaweedfs

**Describe the bug**
for some reason volume server cant delete files from disk after executing command from shell and than mounts volume back
`volume.delete 192.168.0.150:8088 31`

volume server log
`I0420 15:50:09 33501 store.go:369] DeleteVolume 31`
master server
```
I0420 15:50:09  5325 topology.go:184] removing volume info:Id:31, Size:0, ReplicaPlacement:001, Collection:, Version:3, FileCount:0, DeleteCount:0, DeletedByteCount:0, ReadOnly:false
I0420 15:50:09  5325 volume_layout.go:231] Volume 31 becomes unwritable
I0420 15:50:09  5325 master_grpc_server.go:157] master send to filer@192.168.0.190:18888: url:"192.168.0.150:8088" public_url:"192.168.0.150:8088" deleted_vids:31
I0420 15:50:09  5325 master_grpc_server.go:157] master send to filer@192.168.0.191:18888: url:"192.168.0.150:8088" public_url:"192.168.0.150:8088" deleted_vids:31
I0420 15:50:09  5325 master_grpc_server.go:157] master send to master@192.168.0.142:37804: url:"192.168.0.150:8088" public_url:"192.168.0.150:8088" deleted_vids:31
I0420 15:50:09  5325 master_grpc_server.go:157] master send to master@192.168.0.145:42868: url:"192.168.0.150:8088" public_url:"192.168.0.150:8088" deleted_vids:31
I0420 15:50:09  5325 master_grpc_server.go:157] master send to master@192.168.0.146:35731: url:"192.168.0.150:8088" public_url:"192.168.0.150:8088" deleted_vids:31
I0420 15:50:09  5325 master_grpc_server.go:157] master send to shell@192.168.0.190:27890: url:"192.168.0.150:8088" public_url:"192.168.0.150:8088" deleted_vids:31
I0420 15:50:10  5325 volume_layout.go:243] Volume 31 becomes writable
I0420 15:50:10  5325 master_grpc_server.go:111] master see new volume 31 from 192.168.0.150:8088
I0420 15:50:10  5325 master_grpc_server.go:157] master send to filer@192.168.0.191:18888: url:"192.168.0.150:8088" public_url:"192.168.0.150:8088" new_vids:31
I0420 15:50:10  5325 master_grpc_server.go:157] master send to master@192.168.0.142:37804: url:"192.168.0.150:8088" public_url:"192.168.0.150:8088" new_vids:31
I0420 15:50:10  5325 master_grpc_server.go:157] master send to master@192.168.0.145:42868: url:"192.168.0.150:8088" public_url:"192.168.0.150:8088" new_vids:31
I0420 15:50:10  5325 master_grpc_server.go:157] master send to master@192.168.0.146:35731: url:"192.168.0.150:8088" public_url:"192.168.0.150:8088" new_vids:31
I0420 15:50:10  5325 master_grpc_server.go:157] master send to shell@192.168.0.190:27890: url:"192.168.0.150:8088" public_url:"192.168.0.150:8088" new_vids:31
I0420 15:50:10  5325 master_grpc_server.go:157] master send to filer@192.168.0.190:18888: url:"192.168.0.150:8088" public_url:"192.168.0.150:8088" new_vids:31
```
volume info
```
curl "http://192.168.0.146:9333/dir/lookup?volumeId=31&pretty=y"
{
  "volumeId": "31",
  "locations": [
    {
      "url": "192.168.0.147:8088",
      "publicUrl": "192.168.0.147:8088"
    },
    {
      "url": "192.168.0.150:8088",
      "publicUrl": "192.168.0.150:8088"
    }
  ]
}

{
      "Id": 31,
      "Size": 0,
      "ReplicaPlacement": {
        "SameRackCount": 1,
        "DiffRackCount": 0,
        "DiffDataCenterCount": 0
      },
      "Ttl": {
        "Count": 0,
        "Unit": 0
      },
      "Collection": "",
      "Version": 3,
      "FileCount": 0,
      "DeleteCount": 0,
      "DeletedByteCount": 0,
      "ReadOnly": false,
      "CompactRevision": 1,
      "ModifiedAtSecond": 0,
      "RemoteStorageName": "",
      "RemoteStorageKey": ""
    },
```

when i try to move volume to another server
```
volume.move 192.168.0.150:8088 192.168.0.144:8088 36
2020-04-20 15:35:56.888040 I | copying volume 36 from 192.168.0.150:8088 to 192.168.0.144:8088
2020-04-20 15:35:57.235993 I | tailing volume 36 from 192.168.0.150:8088 to 192.168.0.144:8088
2020-04-20 15:36:07.250838 I | deleting volume 36 from 192.168.0.150:8088
2020-04-20 15:36:07.255567 I | moved volume 36 from 192.168.0.150:8088 to 192.168.0.144:8088
```
files remain on disk and now i have 3 copy, neither of them can be deleted from shell
```
{
  "volumeId": "36",
  "locations": [
    {
      "url": "192.168.0.147:8088",
      "publicUrl": "192.168.0.147:8088"
    },
    {
      "url": "192.168.0.144:8088",
      "publicUrl": "192.168.0.144:8088"
    },
    {
      "url": "192.168.0.150:8088",
      "publicUrl": "192.168.0.150:8088"
    }
  ]
}
```
master log
```
I0420 15:35:57  5325 volume_layout.go:231] Volume 36 becomes unwritable
I0420 15:35:57  5325 master_grpc_server.go:157] master send to master@192.168.0.142:37804: url:"192.168.0.144:8088" public_url:"192.168.0.144:8088" new_vids:36
I0420 15:35:57  5325 master_grpc_server.go:157] master send to master@192.168.0.145:42868: url:"192.168.0.144:8088" public_url:"192.168.0.144:8088" new_vids:36
I0420 15:35:57  5325 master_grpc_server.go:157] master send to master@192.168.0.146:35731: url:"192.168.0.144:8088" public_url:"192.168.0.144:8088" new_vids:36
I0420 15:35:57  5325 master_grpc_server.go:157] master send to shell@192.168.0.190:27890: url:"192.168.0.144:8088" public_url:"192.168.0.144:8088" new_vids:36
I0420 15:35:57  5325 master_grpc_server.go:157] master send to filer@192.168.0.190:18888: url:"192.168.0.144:8088" public_url:"192.168.0.144:8088" new_vids:36
I0420 15:35:57  5325 master_grpc_server.go:157] master send to filer@192.168.0.191:18888: url:"192.168.0.144:8088" public_url:"192.168.0.144:8088" new_vids:36
I0420 15:36:07  5325 topology.go:184] removing volume info:Id:36, Size:0, ReplicaPlacement:001, Collection:, Version:3, FileCount:0, DeleteCount:0, DeletedByteCount:0, ReadOnly:false
I0420 15:36:07  5325 volume_layout.go:243] Volume 36 becomes writable
I0420 15:36:07  5325 master_grpc_server.go:157] master send to master@192.168.0.145:42868: url:"192.168.0.150:8088" public_url:"192.168.0.150:8088" deleted_vids:36
I0420 15:36:07  5325 master_grpc_server.go:157] master send to master@192.168.0.146:35731: url:"192.168.0.150:8088" public_url:"192.168.0.150:8088" deleted_vids:36
I0420 15:36:07  5325 master_grpc_server.go:157] master send to shell@192.168.0.190:27890: url:"192.168.0.150:8088" public_url:"192.168.0.150:8088" deleted_vids:36
I0420 15:36:07  5325 master_grpc_server.go:157] master send to filer@192.168.0.190:18888: url:"192.168.0.150:8088" public_url:"192.168.0.150:8088" deleted_vids:36
I0420 15:36:07  5325 master_grpc_server.go:157] master send to filer@192.168.0.191:18888: url:"192.168.0.150:8088" public_url:"192.168.0.150:8088" deleted_vids:36
I0420 15:36:07  5325 master_grpc_server.go:157] master send to master@192.168.0.142:37804: url:"192.168.0.150:8088" public_url:"192.168.0.150:8088" deleted_vids:36
I0420 15:36:10  5325 volume_layout.go:231] Volume 36 becomes unwritable
I0420 15:36:10  5325 master_grpc_server.go:111] master see new volume 36 from 192.168.0.150:8088
I0420 15:36:10  5325 master_grpc_server.go:157] master send to filer@192.168.0.190:18888: url:"192.168.0.150:8088" public_url:"192.168.0.150:8088" new_vids:36
I0420 15:36:10  5325 master_grpc_server.go:157] master send to filer@192.168.0.191:18888: url:"192.168.0.150:8088" public_url:"192.168.0.150:8088" new_vids:36
I0420 15:36:10  5325 master_grpc_server.go:157] master send to master@192.168.0.142:37804: url:"192.168.0.150:8088" public_url:"192.168.0.150:8088" new_vids:36
I0420 15:36:10  5325 master_grpc_server.go:157] master send to master@192.168.0.145:42868: url:"192.168.0.150:8088" public_url:"192.168.0.150:8088" new_vids:36
I0420 15:36:10  5325 master_grpc_server.go:157] master send to master@192.168.0.146:35731: url:"192.168.0.150:8088" public_url:"192.168.0.150:8088" new_vids:36
I0420 15:36:10  5325 master_grpc_server.go:157] master send to shell@192.168.0.190:27890: url:"192.168.0.150:8088" public_url:"192.168.0.150:8088" new_vids:36
I0420 15:40:21  5325 topology.go:184] removing volume info:Id:36, Size:0, ReplicaPlacement:001, Collection:, Version:3, FileCount:0, DeleteCount:0, DeletedByteCount:0, ReadOnly:false
I0420 15:40:21  5325 volume_layout.go:243] Volume 36 becomes writable
I0420 15:40:21  5325 master_grpc_server.go:157] master send to filer@192.168.0.190:18888: url:"192.168.0.144:8088" public_url:"192.168.0.144:8088" deleted_vids:36
I0420 15:40:21  5325 master_grpc_server.go:157] master send to filer@192.168.0.191:18888: url:"192.168.0.144:8088" public_url:"192.168.0.144:8088" deleted_vids:36
I0420 15:40:21  5325 master_grpc_server.go:157] master send to master@192.168.0.142:37804: url:"192.168.0.144:8088" public_url:"192.168.0.144:8088" deleted_vids:36
I0420 15:40:21  5325 master_grpc_server.go:157] master send to master@192.168.0.145:42868: url:"192.168.0.144:8088" public_url:"192.168.0.144:8088" deleted_vids:36
I0420 15:40:21  5325 master_grpc_server.go:157] master send to master@192.168.0.146:35731: url:"192.168.0.144:8088" public_url:"192.168.0.144:8088" deleted_vids:36
I0420 15:40:21  5325 master_grpc_server.go:157] master send to shell@192.168.0.190:27890: url:"192.168.0.144:8088" public_url:"192.168.0.144:8088" deleted_vids:36
I0420 15:40:25  5325 volume_layout.go:231] Volume 36 becomes unwritable
I0420 15:40:25  5325 master_grpc_server.go:111] master see new volume 36 from 192.168.0.144:8088
I0420 15:40:25  5325 master_grpc_server.go:157] master send to filer@192.168.0.191:18888: url:"192.168.0.144:8088" public_url:"192.168.0.144:8088" new_vids:36
I0420 15:40:25  5325 master_grpc_server.go:157] master send to master@192.168.0.142:37804: url:"192.168.0.144:8088" public_url:"192.168.0.144:8088" new_vids:36
I0420 15:40:25  5325 master_grpc_server.go:157] master send to master@192.168.0.145:42868: url:"192.168.0.144:8088" public_url:"192.168.0.144:8088" new_vids:36
I0420 15:40:25  5325 master_grpc_server.go:157] master send to master@192.168.0.146:35731: url:"192.168.0.144:8088" public_url:"192.168.0.144:8088" new_vids:36
I0420 15:40:25  5325 master_grpc_server.go:157] master send to shell@192.168.0.190:27890: url:"192.168.0.144:8088" public_url:"192.168.0.144:8088" new_vids:36
I0420 15:40:25  5325 master_grpc_server.go:157] master send to filer@192.168.0.190:18888: url:"192.168.0.144:8088" public_url:"192.168.0.144:8088" new_vids:36
```
volume server log
```
W0420 15:35:57 16783 volume_info.go:43] unmarshal error: unexpected EOF
I0420 15:35:57 16783 volume_loading.go:109] loading leveldb /mnt/chunk12_sdr/weedfs_volume/36.ldb
I0420 15:35:57 16783 needle_map_leveldb.go:28] Start to Generate /mnt/chunk12_sdr/weedfs_volume/36.ldb from /mnt/chunk12_sdr/weedfs_volume/36.idx
I0420 15:35:57 16783 needle_map_leveldb.go:30] Finished Generating /mnt/chunk12_sdr/weedfs_volume/36.ldb from /mnt/chunk12_sdr/weedfs_volume/36.idx
I0420 15:35:57 16783 disk_location.go:81] data file /mnt/chunk12_sdr/weedfs_volume/36.idx, replicaPlacement=001 v=3 size=8 ttl=
I0420 15:35:57 16783 store.go:315] mount volume 36
I0420 15:40:21 16783 store.go:369] DeleteVolume 36
```

**System Setup**
- List the command line to start
```
/usr/local/bin/weed -logdir=/var/log/seaweedfs -logtostderr=0 master -mdir=/opt/seaweedfs -ip=192.168.0.146 -ip.bind=192.168.0.146 -defaultReplication=001 -port=9333 -peers=192.168.0.142:9333,192.168.0.145:9333,192.168.0.146:9333
/usr/local/bin/weed -logdir=/var/log/seaweedfs volume -max=7,7,7,7,7,7,7 -dataCenter=rinet -dir=/mnt/chunk12_sdu/weedfs_volume,/mnt/chunk12_sdt/weedfs_volume,/mnt/chunk12_sds/weedfs_volume,/mnt/chunk12_sdr/weedfs_volume,/mnt/chunk12_sdq/weedfs_volume,/mnt/chunk12_sdp/weedfs_volume,/mnt/chunk12_sdo/weedfs_volume -ip=192.168.0.144 -ip.bind=192.168.0.144 -mserver=192.168.0.142:9333,192.168.0.145:9333,192.168.0.146:9333 -index=leveldb -port=8088
```
- OS version
debian jessie 8.11 x64
- output of `weed version`
version 30GB 1.71 linux amd64

**Comment by chrislusf** [comment]
try to unmount the volume first, and then delete the volume

**Comment by ignusws** [comment]
> 
> 
> try to unmount the volume first, and then delete the volume

after 
`volume.unmount 192.168.0.150:8088 36`
volume server 
`I0420 17:09:58 33501 store.go:346] UnmountVolume 36`
master server
```
I0420 17:09:58  5325 topology.go:184] removing volume info:Id:36, Size:0, ReplicaPlacement:001, Collection:, Version:3, FileCount:0, DeleteCount:0, DeletedByteCount:0, ReadOnly:false
I0420 17:09:58  5325 volume_layout.go:243] Volume 36 becomes writable
I0420 17:09:58  5325 master_grpc_server.go:157] master send to shell@192.168.0.190:25774: url:"192.168.0.150:8088" public_url:"192.168.0.150:8088" deleted_vids:36
I0420 17:09:58  5325 master_grpc_server.go:157] master send to filer@192.168.0.190:18888: url:"192.168.0.150:8088" public_url:"192.168.0.150:8088" deleted_vids:36
I0420 17:09:58  5325 master_grpc_server.go:157] master send to filer@192.168.0.191:18888: url:"192.168.0.150:8088" public_url:"192.168.0.150:8088" deleted_vids:36
I0420 17:09:58  5325 master_grpc_server.go:157] master send to master@192.168.0.142:37804: url:"192.168.0.150:8088" public_url:"192.168.0.150:8088" deleted_vids:36
I0420 17:09:58  5325 master_grpc_server.go:157] master send to master@192.168.0.145:42868: url:"192.168.0.150:8088" public_url:"192.168.0.150:8088" deleted_vids:36
I0420 17:09:58  5325 master_grpc_server.go:157] master send to master@192.168.0.146:35731: url:"192.168.0.150:8088" public_url:"192.168.0.150:8088" deleted_vids:36
```
then 
`volume.delete 192.168.0.150:8088 36`

nothing logged nor on master nor on volume server, files not deleted on disk

**Comment by chrislusf** [comment]
everything seems right. Could not tell where went wrong. Can not reproduce locally.

**Comment by ignusws** [comment]
> 
> 
> everything seems right. Could not tell where went wrong. Can not reproduce locally.

full cluster restrart didn't help either.

maybe there more verbose logging for volume server that help trace this bug? How to activate it?
is it normal that after deleting attempt volume mounted again?

**Comment by ignusws** [comment]
after some printf's in source code find culprit
here volume.list for node
```
DataNode 192.168.0.144:8088 volume:18/49 active:18 free:31 remote:0
        volume id:1 size:31458512312 collection:"video" file_count:266135 delete_count:27 deleted_byte_count:2252305 replica_placement:1 version:3 modified_at_second:1584679511  
        volume id:3 size:31466205944 collection:"video" file_count:266905 delete_count:46 deleted_byte_count:4895844 replica_placement:1 version:3 modified_at_second:1584679610  
        volume id:6 size:31464731600 collection:"video" file_count:266813 delete_count:41 deleted_byte_count:2987567 replica_placement:1 version:3 modified_at_second:1584679594  
        volume id:7 size:31470414152 collection:"video" file_count:120114 delete_count:7 deleted_byte_count:730568 replica_placement:1 version:3 modified_at_second:1585025283  
        volume id:8 size:31486650112 collection:"video" file_count:119927 delete_count:11 deleted_byte_count:1933257 replica_placement:1 version:3 modified_at_second:1585025292  
        volume id:10 size:31465915784 collection:"video" file_count:119662 delete_count:14 deleted_byte_count:1721282 replica_placement:1 version:3 modified_at_second:1585025283  
        volume id:12 size:31237527240 collection:"video" file_count:119572 delete_count:18 deleted_byte_count:3062165 replica_placement:1 version:3 modified_at_second:1585025303  
        volume id:14 size:31493309880 collection:"video" file_count:102620 delete_count:11 deleted_byte_count:3990014 replica_placement:1 version:3 modified_at_second:1585099796  
        volume id:15 size:31228557496 collection:"video" file_count:102622 delete_count:13 deleted_byte_count:4257759 replica_placement:1 version:3 modified_at_second:1585099764  
        volume id:18 size:31491245080 collection:"video" file_count:102973 delete_count:15 deleted_byte_count:5474686 replica_placement:1 version:3 modified_at_second:1585099802  
        volume id:20 size:31462702296 collection:"video" file_count:101360 delete_count:11 deleted_byte_count:3225382 replica_placement:1 version:3 modified_at_second:1585119775  
        volume id:21 size:31231044128 collection:"video" file_count:101450 delete_count:12 deleted_byte_count:3777483 replica_placement:1 version:3 modified_at_second:1585119756  
        volume id:22 size:31247964560 collection:"video" file_count:101325 delete_count:20 deleted_byte_count:6494369 replica_placement:1 version:3 modified_at_second:1585119804  
        volume id:23 size:31273831712 collection:"video" file_count:101400 delete_count:11 deleted_byte_count:3839289 replica_placement:1 version:3 modified_at_second:1585119825  
        volume id:25 size:7194384248 collection:"video" file_count:25269 replica_placement:1 version:3 modified_at_second:1587436399  
        volume id:26 size:7200137992 collection:"video" file_count:25232 delete_count:3 deleted_byte_count:814410 replica_placement:1 version:3 modified_at_second:1587436399  
        volume id:28 size:7025454808 collection:"video" file_count:24960 delete_count:3 deleted_byte_count:378483 replica_placement:1 version:3 modified_at_second:1587436399  
        volume id:38 size:8 replica_placement:1 version:3 modified_at_second:1587468941  
      DataNode 192.168.0.144:8088 total size:460898589352 file_count:2068339 deleted_file:263 deleted_bytes:49834863
```
and here what return https://github.com/chrislusf/seaweedfs/blob/3775211962f9595e9c00600d23ac841646a7b22a/weed/storage/disk_location.go#L171
l.volumes
```
 volumes map[
1:Id:1, dir:/mnt/chunk12_sdu/weedfs_volume, Collection:video, dataFile:&{0xc0002ba070 /mnt/chunk12_sdu/weedfs_volume/video_1.dat}, nm:&{{{27 266135 2252305 31449860537 3190760} 0xc0002ba078 {0 0}} /mnt/chunk12_sdu/weedfs_volume/video_1.ldb 0xc000592340}, noWrite:false canDelete:false
> 14:Id:14, dir:/mnt/chunk12_sdu/weedfs_volume, Collection:video, dataFile:&{0xc0002ba060 /mnt/chunk12_sdu/weedfs_volume/video_14.dat}, nm:&{{{11 102620 3990014 31489973411 5869548} 0xc0002ba080 {0 0}} /mnt/chunk12_sdu/weedfs_volume/video_14.ldb 0xc000592000}, noWrite:false canDelete:false
> 25:Id:25, dir:/mnt/chunk12_sdu/weedfs_volume, Collection:video, dataFile:&{0xc000010098 /mnt/chunk12_sdu/weedfs_volume/video_25.dat}, nm:&{{{0 25269 0 7193563320 7501149} 0xc0000100e8 {0 0}} /mnt/chunk12_sdu/weedfs_volume/video_25.ldb 0xc0005921a0}, noWrite:false canDelete:false]
```

command line
`/usr/local/bin/weed -logdir=/var/log/seaweedfs -v=5 volume -max=7,7,7,7,7,7,7 -dataCenter=rinet -dir=/mnt/chunk12_sdu/weedfs_volume,/mnt/chunk12_sdt/weedfs_volume,/mnt/chunk12_sds/weedfs_volume,/mnt/chunk12_sdr/weedfs_volume,/mnt/chunk12_sdq/weedfs_volume,/mnt/chunk12_sdp/weedfs_volume,/mnt/chunk12_sdo/weedfs_volume -ip=192.168.0.144 -ip.bind=192.168.0.144 -mserver=192.168.0.142:9333,192.168.0.145:9333,192.168.0.146:9333 -index=leveldb -port=8088`

basically it returns only volumes from first dir and on all other vids function returns nil and stop execution
